components/HamburgerMenu.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+    <div data-testid="hamburger-menu-container" class="cursor-pointer w-full flex" v-close-on-outside-click>
+        <SVGHamburgerMenuIcon data-testid="hamburger-menu-icon" @click="openMenu()" alt="hamburger menu" class="h-8 w-8" />
+        <Transition enter-active-class="transition ease-in-out duration-300 transform: translate(100%, 0)"
+            leave-active-class="transition ease-in-out duration-100 transform: translate(-100%, 0)"
+            enter-from-class="opacity-0 translate-x-2.5" enter-to-class="opacity-100 translate-x-0"
+            leave-from-class="opacity-100 translate-x-0" leave-to-class="opacity-0 translate-x-2.5">
+            <div data-testid="hamburger-menu" v-show="isMenuOpen"
+                class="z-20 absolute top-0 right-0 h-full w-2/3 pt-6 pb-2 rounded bg-primary-bg border-2 flex flex-col justify-between ">
+                <div data-testid="hamburger-upper-section">
+                    <div data-testid="hamburger-header" class="flex justify-between px-5">
+                        <div class="flex">
+                            <div class="title-text flex flex-col flex-shrink-0" data-testid="logo">
+                                <div class="text-lg text-primary-text group-hover:text-primary-hover font-semibold">
+                                    Find a Doc, Japan
```

Comment:
Non-blocking: We should use a variable here instead of raw text—ideally a global one.